piccolo_api/crud/exceptions.py

```diff
@@ -82,7 +86,7 @@ async def inner(*args, **kwargs):
                 },
                 status_code=422,
             )
-        except ForeignKeyViolationError as exception:
+        except (ForeignKeyViolationError, RestrictViolationError) as exception:
```

Comment:
I added `RestrictViolationError` alongside `ForeignKeyViolationError` and the tests passed. Can you check if that is good enough.